src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -214,6 +214,7 @@ def __init__(
         self.expr_to_alias = expr_to_alias if expr_to_alias else {}
         self.session = session
         self.source_plan = source_plan
+        self.children = source_plan.children if source_plan is not None else []
```

Comment:
I see. I know we also added `children` in `SetStatement` and `SelectStatement` in PR #1295 to help us traverse the plan with better accuracy, for same reason we add it here. I'm not sure if we can rely `source_plan` to cover all cases.

Reply:
actually I just realized that we can't just add children for them, because when we resolve a logic plan here https://github.com/snowflakedb/snowpark-python/blob/8a6b98ad662010670f8e4b750fe791c4ba799232/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L727, we always iterate through it's children. So if a SnowflakePlan or Selectable object is passed into this function (I verified Selectable will be passed into it), we will resolve them again, which are unnecessary (though it seems no negative effect). So it'd better to use parameter to protect this change.